azurerm/internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -279,6 +323,7 @@ func resourceArmKustoClusterRead(d *schema.ResourceData, meta interface{}) error
 		d.Set("enable_streaming_ingest", clusterProperties.EnableStreamingIngest)
 		d.Set("enable_purge", clusterProperties.EnablePurge)
 		d.Set("virtual_network_configuration", flatteKustoClusterVNET(clusterProperties.VirtualNetworkConfiguration))
+		d.Set("key_vault", flatteKustoClusterKeyVault(clusterProperties.KeyVaultProperties))
```

Comment:
since this is setting a complex object into the state can we error check this:

```suggestion
		if err := d.Set("key_vault", flatteKustoClusterKeyVault(clusterProperties.KeyVaultProperties)); err != nil {
			return fmt.Errorf("Error setting `key_vault`: %+v", err)
		}
```